### PR TITLE
ZK-3846: use date format in data constraint error messages and locali…

### PR DESCRIPTION
**Pull request by charlesqqq**
…ze range separator

### CODE REVIEW

File: zktest/test/java/org/zkoss/zktest/zats/test2/F85_ZK_3846Test.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+*/
+package org.zkoss.zktest.zats.test2;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.zkoss.zktest.zats.WebDriverTestCase;
+import org.zkoss.zktest.zats.ztl.JQuery;
+
+public class F85_ZK_3846Test extends WebDriverTestCase {
```
Comment (scribetw) [inline zktest/test/java/org/zkoss/zktest/zats/test2/F85_ZK_3846Test.java]: Try `Clients.reloadMessages` for testing de_DE and non de_DE

Comment (charlesqqq) [reply zktest/test/java/org/zkoss/zktest/zats/test2/F85_ZK_3846Test.java]: OK

**Comment by wenninghsu** [comment]
@zkadmin rebase

**Comment by zkadmin** [comment]
I just pushed up the changes, enjoy!

**Pull request merged**